pkg/secrets-store/nodeserver.go

```diff
@@ -118,19 +119,18 @@ func (ns *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 			return nil, status.Errorf(codes.Internal, "failed to check if target path %s is mount point, err: %v", targetPath, err)
 		}
 	}
-	if mounted {
-		klog.InfoS("target path is already mounted", "targetPath", targetPath, "pod", klog.ObjectRef{Namespace: podNamespace, Name: podName})
-		return &csi.NodePublishVolumeResponse{}, nil
-	}
```

Comment:
Is this removed to support `requiresRepublish`? I'm not quite following this part of the change.

If it is to support `requiresRepublish` then I think more work would need to be done so that the secret rotation controller and I'd just want a `TODO` comment mentioning https://github.com/kubernetes-sigs/secrets-store-csi-driver/issues/585 to finish out that.

Reply:
I *believe* this is for the remount requests that get triggered when the token is refreshed. But I agree, we need a `TODO` here with a link to the issue and can remove that when we merge implementation for `requiresRepublish`

Reply:
Yes, this is intended to support `requiresRepublish`. I'll remove this for now and add a TODO

Reply:
I'll keep the `if !mounted {` blocks below since adding this block back will cause them to always be executed